rust/src/interpreter.rs

```diff
@@ -796,10 +722,16 @@ impl<'a> Interpreter<'a> {
                     }
 
                     self.step_status_code = StepStatusCode::EVMC_STEP_STOPPED;
-                    self.code_reader.next();
                     break;
                 }
             }
+
+            if !(Opcode::Push1 as u8..=Opcode::Push32 as u8).contains(&(op as u8))
+                && op != Opcode::Jump
+                && op != Opcode::JumpI
```

Comment:
I am not sure was this implemented, but lfvm does an indiscriminate pc+1 every time. 
Could we clean this up?  

What I don't like is the semantics being separated. One would think that to introduce a new op implementation, we only need to implement the op code, but we have to look at this code a well. 
what do you think about this code.

Reply:
Well for most ops we indeed only have to implement only the op. The only exception are those ops which cause the pc to not be advanced by 1. I don't see any way around this check except for doing the `code_reader.next()` in ever op.